lib/libutee/tee_api_operations.c

```diff
@@ -1118,6 +1118,12 @@ static TEE_Result tee_buffer_update(
 		l = ROUNDUP(op->buffer_offs + slen - buffer_size,
 				op->block_size);
 		l = MIN(op->buffer_offs, l);
+		/*
+		 * If we're buffering only a single block and have a full
+		 * buffer, process it immediately.
+		 */
+		if (!op->buffer_two_blocks && op->buffer_offs == op->block_size)
```

Comment:
op->buffer_offs is guaranteed either op->block_size or 2 * op->block_size at this point, so it should be enough to check op->buffer_two_blocks.

Reply:
Sure, changed.